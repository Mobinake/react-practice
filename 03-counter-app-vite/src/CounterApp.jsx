import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);

	const handleAdd = () => setCounter((c) => c + 1);

	const handleDecrease = () => setCounter((c) => c - 1);

	const handleReset = () => setCounter(() => value);

	return (
		<>
			<h1>CounterApp</h1>
			<h2 data-testid={'counter'}> {counter} </h2>
			<button
				aria-label='btn-add'
				onClick={handleAdd}>
				+1
			</button>
			<button
				aria-label='btn-decrease'
				onClick={handleDecrease}>
				-1
			</button>
			<button
				aria-label='btn-reset'
				onClick={handleReset}>
				Reset
			</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};
