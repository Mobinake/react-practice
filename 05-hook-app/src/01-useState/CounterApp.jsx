import { useState } from 'react';

export const CounterApp = () => {
	const [counters, setCounter] = useState({
		counter1: 10,
		counter2: 20,
		counter3: 30,
	});

	const { counter1, counter2, counter3 } = counters;

	const incrementarCounter1 = () => {
		setCounter({
			...counters,
			counter1: counter1 + 1,
			counter2,
			counter3,
		});
	};
	const incrementarCounter2 = () => {
		setCounter({
			...counters,
			counter2: counter2 + 1,
		});
	};
	const incrementarCounter3 = () => {
		setCounter({
			...counters,

			counter3: counter3 + 1,
		});
	};

	return (
		<>
			<h1>Counter: {counter1}</h1>
			<h1>Counter: {counter2}</h1>
			<h1>Counter: {counter3}</h1>
			<hr />
			<button
				className='btn'
				onClick={() => incrementarCounter1()}>
				+1
			</button>
			<button
				className='btn'
				onClick={() => incrementarCounter2()}>
				+1
			</button>
			<button
				className='btn'
				onClick={() => incrementarCounter3()}>
				+1
			</button>
		</>
	);
};
