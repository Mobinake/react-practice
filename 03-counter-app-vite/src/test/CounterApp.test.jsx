/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../CounterApp';

describe('Pruebas de <CounterApp/>', () => {
	const initialValue = 10;

	test('debe de hacer match con el snapshot', () => {
		const { container } = render(<CounterApp value={initialValue} />);
		expect(container).toMatchSnapshot();
	});

	test('debe de mostrar el valor inicial de 100 <ConuterApp value={100}', () => {
		render(<CounterApp value={initialValue} />);
		expect(screen.getByText(10)).toBeTruthy();
	});

	test('debe de incrementar con el boton +1', () => {
		render(<CounterApp value={initialValue} />);
		fireEvent.click(screen.getByText('+1'));
		expect(screen.getByText('11')).toBeTruthy();
	});

	test('debe de disminuir con el boton -1', () => {
		render(<CounterApp value={initialValue} />);
		fireEvent.click(screen.getByText('-1'));
		expect(screen.getByText('9')).toBeTruthy();
	});

	test('debe de resetear con el boton Reset', () => {
		render(<CounterApp value={initialValue} />);
		fireEvent.click(screen.getByText('+1'));
		fireEvent.click(screen.getByText('+1'));
		fireEvent.click(screen.getByText('+1'));

		// fireEvent.click(screen.getByText('Reset'));
		// expect(screen.getByText('10')).toBeTruthy();

		// con aria-label
		fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
		expect(screen.getByText('10')).toBeTruthy();
	});
});
