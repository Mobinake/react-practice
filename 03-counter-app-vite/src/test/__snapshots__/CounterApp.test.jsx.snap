// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <CounterApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="counter"
  >
     
    10
     
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-decrease"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
